File: solutions/java/space-age/1/src/main/java/SpaceAge.java
class SpaceAge {

    private final double MERCURY_ORBITAL_PERIOD = 0.2408467;
    private final double VENUS_ORBITAL_PERIOD = 0.61519726;
    private final double MARS_ORBITAL_PERIOD = 1.8808158;
    private final double JUPITER_ORBITAL_PERIOD = 11.862615;
    private final double SATURN_ORBITAL_PERIOD = 29.447498;
    private final double URANUS_ORBITAL_PERIOD = 84.016846;
    private final double NEPTUNE_ORBITAL_PERIOD = 164.79132;

    private final double ageInSeconds;

    private final double secondsPerYear = 365.25 * 24 * 60 * 60;

    SpaceAge(double seconds) {
        this.ageInSeconds = seconds;
    }

    double getSeconds() {
        return ageInSeconds;
    }

    double onEarth() {
        return ageInSeconds / secondsPerYear;
    }

    double onMercury() {
        return ageInSeconds / (secondsPerYear * MERCURY_ORBITAL_PERIOD);
    }

    double onVenus() {
        return ageInSeconds / (secondsPerYear * VENUS_ORBITAL_PERIOD);
    }

    double onMars() {
        return ageInSeconds / (secondsPerYear * MARS_ORBITAL_PERIOD);
    }

    double onJupiter() {
        return ageInSeconds / (secondsPerYear * JUPITER_ORBITAL_PERIOD);
    }

    double onSaturn() {
        return ageInSeconds / (secondsPerYear * SATURN_ORBITAL_PERIOD);
    }

    double onUranus() {
        return ageInSeconds / (secondsPerYear * URANUS_ORBITAL_PERIOD);
    }

    double onNeptune() {
        return ageInSeconds / (secondsPerYear * NEPTUNE_ORBITAL_PERIOD);
    }

}

File: solutions/csharp/space-age/1/SpaceAge.cs
public class SpaceAge(int seconds)
{

    private const double EarthYearDays = 365.25f;

    private double _earthAge = seconds / (EarthYearDays * 24 * 60 * 60);

    private const double MercuryOrbitalPeriod =  0.2408467;
    private const double VenusOrbitalPeriod =  0.61519726;
    private const double MarsOrbitalPeriod =  1.8808158;
    private const double JupiterOrbitalPeriod =  11.862615;
    private const double SaturnOrbitalPeriod =  29.447498;
    private const double UranusOrbitalPeriod =  84.016846;
    private const double NeptuneOrbitalPeriod =  164.79132;
    
    public double OnEarth() => _earthAge;

    public double OnMercury() => _earthAge / MercuryOrbitalPeriod;

    public double OnVenus() => _earthAge / VenusOrbitalPeriod;

    public double OnMars() => _earthAge / MarsOrbitalPeriod;

    public double OnJupiter() => _earthAge / JupiterOrbitalPeriod;

    public double OnSaturn() => _earthAge / SaturnOrbitalPeriod;

    public double OnUranus() => _earthAge / UranusOrbitalPeriod;

    public double OnNeptune() => _earthAge / NeptuneOrbitalPeriod;
}